akka-stream-tests/src/test/java/akka/stream/javadsl/SourceTest.java

```diff
@@ -966,21 +951,26 @@ public void cycleSourceMustGenerateSameSequenceInRepeatedFashion() throws Except
     // #cycle
   }
 
-  @Test(expected = IllegalArgumentException.class)
-  public void cycleSourceMustThrow() throws Throwable {
-
-    try {
-      // #cycle-error
-      Iterator<Integer> emptyIterator = Collections.<Integer>emptyList().iterator();
-      Source.cycle(() -> emptyIterator)
-          .runWith(Sink.head(), system)
-          // stream will be terminated with IllegalArgumentException
-          // #cycle-error
-          .toCompletableFuture()
-          .get();
-    } catch (ExecutionException e) {
-      throw e.getCause();
-    }
+  @Test
+  public void cycleSourceMustThrow() {
+    ExecutionException exception =
+        Assert.assertThrows(
```

Comment:
Perhaps do a static import of this like with other static assert methods.

Reply:
It's the cool suggestion, but I couldn't. `JUnitSuiteLike` has own `assertThrows` (different signature). I had compilation exception even with static import. I should specified the full name.  Am I wrong?

Reply:
Ah, thanks for explaining, let's merge as is then :)